symbolic_pymc/tensorflow/printing.py

```diff
@@ -58,41 +67,41 @@ def tf_dprint(obj, printer=None):
             )
 
     if printer is None:
-        printer = TFlowPrinter(str, sys.stdout)
+        printer = TFlowPrinter(str, sys.stdout, index)
 
-    _tf_dprint(obj, printer)
+    _tf_dprint(obj, printer, depth=0)
 
 
 @singledispatch
-def _tf_dprint(obj, printer):
-    printer.println(obj)
+def _tf_dprint(obj, printer, depth):
+    if printer.depth_check(depth):
```

Comment:
Why have the checking logic in the `TFlowPrinter` object when, especially when the depth value is exclusively managed *outside* the `TFlowPrinter` object?

You could probably track depth in the same way/alongside as the indentation level, or&mdash;better yet&mdash;you could change the indentation tracking to depth tracking (if only in naming) and reasonably double-purpose that.

Reply:
I took your advice and renamed `self.indentation` to `self.depth_tracking`, the functionality is still the same and depth is now calculated based off this.